test/blackbox-tests/test-cases/pkg/env-conditional-dependencies.t

```diff
@@ -25,6 +25,7 @@ dependencies are included.
   $ solve foo
   Solution for dune.lock:
   - foo.0.0.1
+  - foo-linux.0.0.1
```

Comment:
this extra dependency looks suspicious. Do you need to unset `os` in the solver env for this test now?

Reply:
Ah yes, this test no longer makes sense. I added a lock dir with `os` unset.